### PR TITLE
Avoid exceptions when authenticating users with an invalid encoding

### PR DESCRIPTION
**Pull request by segiddins**
Fixes https://app.honeybadger.io/projects/40972/faults/102413552

This is a steady source of exceptions for us, this should fail more gracefully

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -72,9 +72,13 @@ class User < ApplicationRecord
   validate :toxic_email_domain, on: :create
 
   def self.authenticate(who, password)
+    # Avoid exceptions when string is invalid in the given encoding, _or_ cannot be converted
+    # to UTF-8.
+    who = who.encode(Encoding::UTF_8)
```
Comment (simi) [inline app/models/user.rb:77]: :thinking: Is this the only place facing this problem? Isn't this related to almost any input?

Comment (segiddins) [reply app/models/user.rb:77]: This is one of the places it shows up most often, because we are not using controller params to get these

Comment (simi) [reply app/models/user.rb:77]: Clear, wouldn't make sense to keep this together in controller than?

**Review by simi — APPROVED** [review]
I was looking at alternative solutions, but there is really nothing systematic and simple at the same time to do in here.

**Pull request merged**